lib/src/signed_video_h26x_common.c

```diff
@@ -843,37 +843,35 @@ update_gop_hash(void *crypto_handle, gop_info_t *gop_info)
 
   return status;
 }
-svrc_t
-compute_partial_gop_hash(const signed_video_t *self,
-    const uint8_t *hash_list,
-    int hash_list_idx,
-    uint8_t *gop_hash)
-{
-  size_t hash_size = openssl_get_hash_size(self->crypto_handle);
-  if (hash_list_idx <= 0) {
-    memset(gop_hash, 0, hash_size);
-    return SV_OK;  // Handle insufficient memory scenario.
-  }
-
-  return openssl_hash_data(self->crypto_handle, hash_list, hash_list_idx, gop_hash);
-}
 
-/* Checks if there is enough room to copy the hash. If so, copies the |nalu_hash| and updates the
- * |list_idx|. Otherwise, sets the |list_idx| to -1 and proceeds. */
+/* Initializes and updates the GOP hash regardless of available space. If there is enough
+ * room, copies the |hash| and updates |list_idx|. Otherwise, sets |list_idx| to -1.
+ */
 void
 check_and_copy_hash_to_hash_list(signed_video_t *self, const uint8_t *hash, size_t hash_size)
 {
   if (!self || !hash) return;
 
   uint8_t *hash_list = &self->gop_info->hash_list[0];
   int *list_idx = &self->gop_info->list_idx;
-  // Check if there is room for another hash in the |hash_list|.
-  if (*list_idx + hash_size > self->gop_info->hash_list_size) *list_idx = -1;
+
+  // If the upcoming hash doesn't fit in the hash list buffer, set *list_idx to -1
+  // to indicate that the hash list is full, and the hash list is no longer accessible.
+  if (*list_idx + hash_size > self->gop_info->hash_list_size) {
+    *list_idx = -1;
+  }
+  // If this is the start of the GOP, initialize |crypto_handle| to enable
+  // updating the hash with each received NALU.
+  if (*list_idx == 0) {
```

Comment:
For example, if `self->gop_info->hash_list_size = 32` and `hash_size = 64` the GOP hash will never be initialized. This is another example on why it is good to keep the hash list and computation of the GOP hash separated.